core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/ArtifactSignerFactory.java

```diff
@@ -12,179 +12,32 @@
  */
 package tech.pegasys.eth2signer.core.multikey.metadata;
 
-import static java.lang.String.format;
-import static java.nio.charset.StandardCharsets.UTF_8;
-
-import tech.pegasys.eth2signer.core.metrics.Eth2SignerMetricCategory;
 import tech.pegasys.eth2signer.core.signing.ArtifactSigner;
-import tech.pegasys.eth2signer.core.signing.BlsArtifactSigner;
-import tech.pegasys.signers.azure.AzureKeyVault;
-import tech.pegasys.signers.bls.keystore.KeyStore;
-import tech.pegasys.signers.bls.keystore.KeyStoreLoader;
-import tech.pegasys.signers.bls.keystore.KeyStoreValidationException;
-import tech.pegasys.signers.bls.keystore.model.KeyStoreData;
-import tech.pegasys.signers.hashicorp.HashicorpConnection;
-import tech.pegasys.signers.hashicorp.HashicorpConnectionFactory;
-import tech.pegasys.signers.hashicorp.TrustStoreType;
-import tech.pegasys.signers.hashicorp.config.ConnectionParameters;
-import tech.pegasys.signers.hashicorp.config.KeyDefinition;
-import tech.pegasys.signers.hashicorp.config.TlsOptions;
-import tech.pegasys.teku.bls.BLSKeyPair;
-import tech.pegasys.teku.bls.BLSSecretKey;
-
-import java.io.FileNotFoundException;
-import java.io.IOException;
-import java.nio.file.Path;
-import java.util.Optional;
-
-import com.google.common.io.Files;
-import org.apache.tuweni.bytes.Bytes;
-import org.hyperledger.besu.plugin.services.MetricsSystem;
-import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
-import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
-import org.hyperledger.besu.plugin.services.metrics.OperationTimer.TimingContext;
 
-public class ArtifactSignerFactory {
+public interface ArtifactSignerFactory {
 
-  private final LabelledMetric<OperationTimer> privateKeyRetrievalTimer;
-  private final Path configsDirectory;
-  private final HashicorpConnectionFactory connectionFactory;
-
-  public ArtifactSignerFactory(
-      final Path configsDirectory,
-      final MetricsSystem metricsSystem,
-      final HashicorpConnectionFactory connectionFactory) {
-    this.configsDirectory = configsDirectory;
-    privateKeyRetrievalTimer =
-        metricsSystem.createLabelledTimer(
-            Eth2SignerMetricCategory.SIGNING,
-            "private_key_retrieval_time",
-            "Time taken to retrieve private key",
-            "signer");
-    this.connectionFactory = connectionFactory;
+  default ArtifactSigner create(FileRawSigningMetadata fileRawSigningMetadata) {
+    throw new UnsupportedOperationException(
+        "Unable to generate a signer of requested type from supplied metadata");
```

Comment:
interesting approach. I wonder if we can somehow use generic T and use one method. Anyways, it looks fine for now 👍